src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/MetadataReaderExtensions.Ecma.cs

```diff
@@ -78,5 +86,257 @@ public static RuntimeAssemblyName ToRuntimeAssemblyName(this AssemblyReferenceHa
+        public static bool GetAttributeNamespaceAndName(MetadataReader metadataReader, EntityHandle typeDefOrRef, out StringHandle namespaceHandle, out StringHandle nameHandle)
+        {
+            nameHandle = default(StringHandle);
+            namespaceHandle = default(StringHandle);
+
+            if (typeDefOrRef.Kind == HandleKind.TypeReference)
+            {
+                TypeReference typeRefRow = metadataReader.GetTypeReference((TypeReferenceHandle)typeDefOrRef);
+                HandleKind handleType = typeRefRow.ResolutionScope.Kind;
+
+                if (handleType == HandleKind.TypeReference || handleType == HandleKind.TypeDefinition)
+                {
+                    // TODO - Support nested types.  
+                    return false;
+                }
+
+                nameHandle = typeRefRow.Name;
+                namespaceHandle = typeRefRow.Namespace;
+            }
+            else if (typeDefOrRef.Kind == HandleKind.TypeDefinition)
+            {
+                var def = metadataReader.GetTypeDefinition((TypeDefinitionHandle)typeDefOrRef);
+
+                if (IsNested(def.Attributes))
+                {
+                    // TODO - Support nested types. 
+                    return false;
+                }
+
+                nameHandle = def.Name;
+                namespaceHandle = def.Namespace;
+            }
+            else
+            {
+                // unsupported metadata
+                throw new BadImageFormatException();
+            }
+
+            return true;
+        }
+
+
+        public static bool IsNested(TypeAttributes flags)
+        {
+            return (flags & ((TypeAttributes)0x00000006)) != 0;
+        }
+
+        internal enum CorElementType : byte
+        {
+            Invalid,
+            ELEMENT_TYPE_VOID,
+            ELEMENT_TYPE_BOOLEAN,
+            ELEMENT_TYPE_CHAR,
+            ELEMENT_TYPE_I1,
+            ELEMENT_TYPE_U1,
+            ELEMENT_TYPE_I2,
+            ELEMENT_TYPE_U2,
+            ELEMENT_TYPE_I4,
+            ELEMENT_TYPE_U4,
+            ELEMENT_TYPE_I8,
+            ELEMENT_TYPE_U8,
+            ELEMENT_TYPE_R4,
+            ELEMENT_TYPE_R8,
+            ELEMENT_TYPE_STRING,
+            ELEMENT_TYPE_PTR,
+            ELEMENT_TYPE_BYREF,
+            ELEMENT_TYPE_VALUETYPE,
+            ELEMENT_TYPE_CLASS,
+            ELEMENT_TYPE_VAR,
+            ELEMENT_TYPE_ARRAY,
+            ELEMENT_TYPE_GENERICINST,
+            ELEMENT_TYPE_TYPEDBYREF,
+            ELEMENT_TYPE_I = 24,
+            ELEMENT_TYPE_U,
+            ELEMENT_TYPE_FNPTR = 27,
+            ELEMENT_TYPE_OBJECT,
+            ELEMENT_TYPE_SZARRAY,
+            ELEMENT_TYPE_MVAR,
+            ELEMENT_TYPE_CMOD_REQD,
+            ELEMENT_TYPE_CMOD_OPT,
+            ELEMENT_TYPE_HANDLE = 64,
+            ELEMENT_TYPE_SENTINEL,
+            ELEMENT_TYPE_PINNED = 69
+        }
+
+        public static void SkipType(ref BlobReader reader)    
```

Comment:
BlobReading code might be better to have in its own class rather than MetadataExtensions (especially since these aren't extension methods.)